custom_components/zwave_mqtt/climate.py

```diff
@@ -0,0 +1,448 @@
+        """Return the fan speed set."""
+        if self._current_fan_mode is None:
+            return None
+        return self._fan_value_label_mapping.get(self._current_fan_mode)
+
+    @property
+    def fan_modes(self):
+        """Return a list of available fan modes."""
+        if not self._fan_modes:
+            return None
+        fan_mode_labels = []
+        for mode_value in self._fan_modes:
+            fan_mode_labels.append(self._fan_value_label_mapping.get(mode_value))
+        return fan_mode_labels
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        if self._unit == "C":
+            return TEMP_CELSIUS
+        if self._unit == "F":
+            return TEMP_FAHRENHEIT
+        return self._unit
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def hvac_mode(self):
+        """Return hvac operation ie. heat, cool mode."""
+        """Needs to be one of HVAC_MODE_*."""
+        if not self._mode():
+            return self._default_hvac_mode
+        return HVAC_MODE_MAPPINGS.get(self._hvac_mode)
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac operation modes."""
+        """Need to be a subset of HVAC_MODES."""
+        if not self._mode():
+            return []
+        return self._hvac_list
+
+    @property
+    def hvac_action(self):
+        """Return the current running hvac operation if supported."""
+        """Needs to be one of CURRENT_HVAC_*."""
+        return self._hvac_action
+
+    @property
+    def preset_mode(self):
+        """Return preset operation ie. eco, away."""
+        """Needs to be one of PRESET_*."""
+        if not self._mode() or self._preset_mode == PRESET_NONE:
+            return PRESET_NONE
+        return self._hvac_value_label_mapping.get(self._preset_mode)
+
+    @property
+    def preset_modes(self):
+        """Return the list of available preset operation modes."""
+        """Need to be a subset of PRESET_MODES."""
+        if not self._mode():
+            return []
+        preset_labels = []
+        for value in self._preset_list:
+            preset_labels.append(self._hvac_value_label_mapping.get(value))
+        preset_labels.append(PRESET_NONE)
+        return preset_labels
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the lowbound target temperature we try to reach."""
+        return self._target_temperature_range[0]
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the highbound target temperature we try to reach."""
+        return self._target_temperature_range[1]
+
+    def set_temperature(self, **kwargs):
```

Comment:
We should use the async api in the whole platform.
```suggestion
    async def async_set_temperature(self, **kwargs):
```

Reply:
Is it planned to convert all the various components to async in the future? I don't think the other components are async. I'm also unfamiliar with the ``send_value`` method and it asynchronicity.

Reply:
What components do you refer too? If you look at the other existing platforms in this integration they all use the async api for the actuator methods etc.

Reply:
Hmm, it seems the fan platform doesn't do that. We should fix that (not here).

Reply:
`send_value` is a callback that should run in the event loop since the `send_message` callback should run in the event loop.

Reply:
Fixed the fan platform here:
https://github.com/cgarwood/homeassistant-zwave_mqtt/pull/107